stdlib/public/core/Collection.swift

```diff
@@ -1003,6 +1022,54 @@ extension Collection {
     return count
   }
 
+  /// Returns a random element of the collection, using the given generator as
+  /// a source for randomness.
+  ///
+  /// You use this method to select a random element from a collection when you
+  /// are using a custom random number generator. For example, call
+  /// `randomElement(using:)` to select a random element from an array of names.
+  ///
+  ///     let names = ["Zoey", "Chloe", "Amani", "Amaia"]
+  ///     let randomName = names.randomElement(using: &myGenerator)!
+  ///     // randomName == "Amani" (maybe)
+  ///
+  /// - Parameter generator: The random number generator to use when choosing
+  ///   a random element.
+  /// - Returns: A random element from the collection. If the collection is
+  ///   empty, the method returns `nil`.
+  @inlinable
+  public func randomElement<T: RandomNumberGenerator>(
+    using generator: inout T
+  ) -> Element? {
+    guard !isEmpty else { return nil }
+    let random = generator.next(upperBound: UInt(count))
+    let index = self.index(
+      startIndex,
+      offsetBy: numericCast(random)
```

Comment:
However, this would be indirectly calling `Range.randomElement(using:)` in the default implementation of `Collection.randomElement(using:)`, which is a protocol requirement.

Reply:
Indirectly calling `Range.randomElement(using:)` inside `Collection.randomElement(using:)` won't be recursive. But will it be slower than `next(upperBound:)` and `numericCast(_:)`?

Reply:
My initial reaction is yes it will be slower, but it should be minuscule however.